app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -1,12 +1,168 @@
 package org.schabi.newpipe.settings;
 
+import android.app.Activity;
 import android.os.Bundle;
+import android.support.v7.preference.ListPreference;
+import android.support.v7.preference.Preference;
 
 import org.schabi.newpipe.R;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.report.ErrorActivity;
+import org.schabi.newpipe.report.UserAction;
+import org.schabi.newpipe.util.Constants;
+import org.schabi.newpipe.util.KioskTranslator;
 
 public class ContentSettingsFragment extends BasePreferenceFragment {
+
     @Override
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+
         addPreferencesFromResource(R.xml.content_settings);
+
+        final ListPreference mainPageContentPref =  (ListPreference) findPreference(getString(R.string.main_page_content_key));
+        mainPageContentPref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+                    @Override
+                    public boolean onPreferenceChange(Preference preference, Object newValueO) {
+                        final String newValue = newValueO.toString();
+
+                        final String mainPrefOldValue =
+                                defaultPreferences.getString(getString(R.string.main_page_content_key), "blank_page");
+                        final String mainPrefOldSummary = getMainPagePrefSummery(mainPrefOldValue, mainPageContentPref);
+
+                        if(newValue.equals(getString(R.string.kiosk_page_key))) {
+                            SelectKioskFragment selectKioskFragment = new SelectKioskFragment();
+                            selectKioskFragment.setOnSelectedLisener(new SelectKioskFragment.OnSelectedLisener() {
+                                @Override
+                                public void onKioskSelected(String kioskId, int service_id) {
+                                    defaultPreferences.edit()
+                                            .putInt(getString(R.string.main_page_selected_service), service_id).apply();
+                                    defaultPreferences.edit()
+                                            .putString(getString(R.string.main_page_selectd_kiosk_id), kioskId).apply();
+                                    String summary = "";
```

Comment:
Hard-coded string format. A better option would be to use string resource with the value 
`%1$s/%2$s`

Reply:
done